packages/connect-core/src/errors.ts

```diff
@@ -0,0 +1,168 @@
+function formatMessage(message: string, reason: string): string {
+  return message + reason ? ` (reason: ${reason})` : ''
+}
+
+function defineNonEnumerable(instance: object, name: string, value: any) {
+  Object.defineProperty(instance, name, { value, enumerable: false })
```

Comment:
This is interesting, I had to read the documentation on `Object.defineProperty` to better understand what you were doing here. I notice the `enumerable` default value is `false. But I guess better to be explicit.

Reply:
Ah yes, this is all so that these properties don’t appear in the console when an error is thrown :smile: